### DISCUSSION TITLE
API rule for authenticated users

### DISCUSSION THREAD

**Discussion by jnatmorris** [pocketbase/pocketbase#1861] [Q&A] [answered]
Quick question, how does one only allow authenticated users to list a collection of records? 

- PocketBase version: v0.12.3
- Method of retrieving records: GET request

From reading the docs and intuition, I thought `@request.auth.id != ""` would provide me with a method of only allowing authenticated users to list the collection of records, but I'm still able to get back results. I'm guessing there is a method for only allowing authenticated users to make API calls by passing a token or such. I know this is possible when you're an admin, but do I have to make all users admins...?

While I have put the code that makes the GET request behind a middleware that enforces authentication, I wanted to look to put rules directly on the database too as theoretically, from a bad actor, requests could be made from outside the purview middleware too.

If anyone has any insights, I'm open to hear them.

Cheers!

**Comment by ganigeorgiev** [comment, marked as answer]
> I thought @request.auth.id != "" would provide me with a method of only allowing authenticated users to list the collection of records, but I'm still able to get back results.

`@request.auth.id != ""` will require the request to have valid `Authorization` token sent with it (aka. the client to be authenticated). If the client is not authenticated, you'll still get a valid 200 response for the list/search action but with empty `items` array.

If you are not experiencing the above behavior, then something else is happening in your app (it could be the middleware you've mentioned or something else).

To rule out PocketBase issue, you can try making the `GET /api/collections/yourCollection/records` request with curl or a rest client (Postman, Insomnia, etc), or even just accessing the API url in the browser (eg. `http://127.0.0.1:8090/api/collections/yourCollection/records`).

**Reply by jnatmorris** [reply]
I can confirm this is working as expected. As I had no elements in the collection, and was getting back an empty array with 200 response, I thought this was wrong. Now that I entered a single record, I'm seeing the difference between being authenticated and not.

To streamline, maybe this can be show in _API Preview_ under the appropriate actions, "List/Search", "View", ect. if an authorization token is required. Or rather than getting an empty 200 array, a 403 with the message about needing an authentication token provided could be added.

**Reply by ganigeorgiev** [reply]
> Or rather than getting an empty 200 array, a 403 with the message about needing an authentication token provided could be added.

We can't do this because all API rules act also as filter, aka. you can have a rule like:
```js
@request.auth.id != "" || status = "public"
```
